src/libraries/System.Text.Json/gen/JsonSourceGenerator.Roslyn4.0.cs

```diff
@@ -27,7 +29,9 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
         {
             IncrementalValuesProvider<ClassDeclarationSyntax> classDeclarations = context.SyntaxProvider
                 .ForAttributeWithMetadataName(
+#if !ROSLYN4_4_OR_GREATER
                     context,
+#endif
```

Comment:
why do we still polyfill here?

Reply:
This file is used in the generator that runs on downlevel versions of VS (so if someone uses System.Text.Json in a .NET Standard 2.0 project, they can still use the source-generated code in VS 17.x where `x<4` or with the .NET 6 SDK and get better perf than the V1 generator APIs provide)

Reply:
Cool, thanks! :)